### PR TITLE
[DOC] Improve the documentation of get_clusters_table

### LINKED ISSUE #3176: `nilearn.reporting.get_cluster_table` poorly documented
It should be clear whether one uses p-values or z-values for thresholding, or there should be a kw for users to control it.

### Affected pages

`nilearn.reporting.get_cluster_table`

### PR DESCRIPTION
**Pull request by tsalo**
Closes #3176.

Changes proposed in this pull request:

- Add information about what kinds of values are valid inputs to `get_clusters_table`.
- Improve the docstring of `get_clusters_table` and private functions in `nilearn.reporting._get_clusters_table`.

### CODE REVIEW

File: nilearn/reporting/_get_clusters_table.py
Comment (NicolasGensollen) [inline nilearn/reporting/_get_clusters_table.py:179]: Maybe you could add the return type for consistency here.

File: nilearn/reporting/_get_clusters_table.py
Comment (NicolasGensollen) [inline nilearn/reporting/_get_clusters_table.py:212]:
```suggestion
        Cluster size threshold, in :term:`voxels<voxel>`. If None, then no cluster size
```

File: nilearn/reporting/_get_clusters_table.py
Comment (NicolasGensollen) [inline nilearn/reporting/_get_clusters_table.py:225]: Could you cross reference the DataFrame return type?

File: nilearn/reporting/_get_clusters_table.py
Comment (NicolasGensollen) [inline nilearn/reporting/_get_clusters_table.py:263]: I might be wrong, but I don't think this is equivalent to the previous code:

```python
>>> ndimage.generate_binary_structure(rank=3, connectivity=2).astype(int)
array([[[0, 1, 0],
        [1, 1, 1],
        [0, 1, 0]],

       [[1, 1, 1],
        [1, 1, 1],
        [1, 1, 1]],

       [[0, 1, 0],
        [1, 1, 1],
        [0, 1, 0]]])
```

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @tsalo this looks great ! 👍 
I have a few small suggestions and a question (see below).
Also, could you add a whatsnew entry ?

File: nilearn/reporting/_get_clusters_table.py
Comment (tsalo) [reply nilearn/reporting/_get_clusters_table.py:263]: You're right, thank you for catching that. I copied this from another repo where I'm using NN2 instead of NN1.

Comment (tsalo) [reply nilearn/reporting/_get_clusters_table.py:263]:
```suggestion
    conn_mat = ndimage.generate_binary_structure(rank=3, connectivity=1)
```

File: nilearn/reporting/_get_clusters_table.py
Comment (tsalo) [inline nilearn/reporting/_get_clusters_table.py:224]:
```suggestion
    df : :obj:`pandas.DataFrame`
```

File: nilearn/reporting/_get_clusters_table.py
Comment (tsalo) [reply nilearn/reporting/_get_clusters_table.py:179]:
```suggestion
    :obj:`float` or :obj:`int`
```

File: nilearn/reporting/_get_clusters_table.py
Comment (tsalo) [reply nilearn/reporting/_get_clusters_table.py]:
```python
import numpy as np
from scipy import ndimage

conn_mat = np.zeros((3, 3, 3), int)
conn_mat[1, 1, :] = 1
conn_mat[1, :, 1] = 1
conn_mat[:, 1, 1] = 1
conn_mat = conn_mat.astype(bool)

conn_mat_new_bad = ndimage.generate_binary_structure(rank=3, connectivity=2)
print(np.array_equal(conn_mat, conn_mat_new_bad))
# False
conn_mat_new = ndimage.generate_binary_structure(rank=3, connectivity=1)
print(np.array_equal(conn_mat, conn_mat_new))
# True
```

**Review by bthirion — APPROVED** [review]
LGTM, thx !

**Comment by tsalo** [comment]
I think this just needs one more approval before it can be merged.

**Review by NicolasGensollen — APPROVED** [review]
Sorry I forgot to approve...
LGTM, thanks @tsalo !

**Pull request merged**
